framework/src/base/EigenProblem.C

```diff
@@ -17,19 +17,29 @@
 #include "EigenProblem.h"
 #include "DisplacedProblem.h"
 #include "Assembly.h"
+#include "SlepcSupport.h"
+
+// libmesh includes
+#include "libmesh/system.h"
+#include "libmesh/eigen_solver.h"
 
 template<>
 InputParameters validParams<EigenProblem>()
 {
   InputParameters params = validParams<FEProblemBase>();
   params.addParam<unsigned int>("n_eigen_pairs", 1, "The dimension of the nullspace");
   params.addParam<unsigned int>("n_basis_vectors", 3, "The dimension of the nullspace");
+  params.addParam<bool>("generalized_eigenvalue_problem", false, "if we are trying to solve a generalized eigenvalue problem");
+#if LIBMESH_HAVE_SLEPC
+  params += Moose::SlepcSupport::getSlepcValidParams();
```

Comment:
And the previous parameter `generalized_eigenvalue_problem` looks redundant with the SLEPc parameter `eigenvalue_problem_type`.